internal/rotate_est.go

```diff
@@ -0,0 +1,207 @@
+package internal
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/ThalesIgnite/crypto11"
+	"github.com/miekg/pkcs11"
+	"go.mozilla.org/pkcs7"
+)
+
+var (
+	oidKeyUsage         = asn1.ObjectIdentifier([]int{2, 5, 29, 15})
+	oidExtendedKeyUsage = asn1.ObjectIdentifier([]int{2, 5, 29, 37})
+
+	asn1DigitalSignature = []byte{3, 2, 7, 128}
+	asn1TlsWebClientAuth = []byte{48, 10, 6, 8, 43, 6, 1, 5, 5, 7, 3, 2}
```

Comment:
Just IMHO what you probably need here is to reference to the standard library code which explains these bits:
```
    // Copied from the crypto/x509/x509.go
    // x509.oidExtensionKeyUsage
    oidKeyUsage         = asn1.ObjectIdentifier([]int{2, 5, 29, 15})

    // x509.oidExtensionExtendedKeyUsage
    oidExtendedKeyUsage = asn1.ObjectIdentifier([]int{2, 5, 29, 37})

    // x509.marshalKeyUsage(x509.KeyUsageDigitalSignature)
    asn1DigitalSignature = []byte{3, 2, 7, 128}

    // x509.marshalExtKeyUsage([]x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth}, []asn1.ObjectIdentifier{})
    // or below (if you like it more)
    // asn1.Marshal([]asn1.ObjectIdentifier{x509.oidExtKeyUsageClientAuth})
    asn1TlsWebClientAuth = []byte{48, 10, 6, 8, 43, 6, 1, 5, 5, 7, 3, 2}    
```

That said, I'm once again convinced that the universe hates software engineers by inventing ASN1 and X509.